integration/query_comparision_test.go

```diff
@@ -39,22 +39,37 @@ func TestQueryComparisonImplicit(t *testing.T) {
 		filter      bson.D
 		expectedIDs []any
 	}{
+		"Document": {
+			filter:      bson.D{{"value", bson.D{{"foo", int32(42)}, {"42", "foo"}, {"array", bson.A{int32(42), "foo", nil}}}}},
```

Comment:
Let's also add a test for comparision with document with the same keys and values, but in different order. The same in other tests

Reply:
Will do.

Reply:
Done